python/cells.cpp

```diff
@@ -0,0 +1,226 @@
+                    dend->add_mechanism("pas");
+                    dend->rL = 100;
+                }
+            }
+        }
+        if (sec_ids.empty()) {
+            break;
+        }
+        levels.push_back(sec_ids);
+    }
+
+    // Add spike threshold detector at the soma.
+    cell.add_detector({0,0}, 10);
+
+    // Add a synapse to the mid point of the first dendrite.
+    cell.add_synapse({1, 0.5}, "expsyn");
+
+    // Add additional synapses that will not be connected to anything.
+    for (unsigned i=1u; i<params.synapses; ++i) {
+        cell.add_synapse({1, 0.5}, "expsyn");
+    }
+
+    return cell;
+}
+
+void register_cells(pybind11::module& m) {
+    using namespace pybind11::literals;
+
+    pybind11::class_<arb::spike_source_cell> spike_source_cell(m, "spike_source_cell",
+        "A spike source cell, that generates a user-defined sequence of spikes that act as inputs for other cells in the network.");
+
+    spike_source_cell
+        .def(pybind11::init<>(
+            [](const regular_schedule_shim& sched){
+                return arb::spike_source_cell{sched.schedule()};}),
+            "schedule"_a, "Construct a spike source cell that generates spikes at regular intervals")
+        .def(pybind11::init<>(
+            [](const explicit_schedule_shim& sched){
+                return arb::spike_source_cell{sched.schedule()};}),
+            "schedule"_a, "Construct a spike source cell that generates spikes at a sequence of user-defined times")
+        .def(pybind11::init<>(
+            [](const poisson_schedule_shim& sched){
+                return arb::spike_source_cell{sched.schedule()};}),
+            "schedule"_a, "Construct a spike source cell that generates spikes at times defined by a Poisson sequence")
+        .def("__repr__", [](const arb::spike_source_cell&){return "<spike_source_cell>";})
+        .def("__str__",  [](const arb::spike_source_cell&){return "<spike_source_cell>";});
+
+    pybind11::class_<arb::benchmark_cell> benchmark_cell(m, "benchmark_cell",
+        "A benchmarking cell, used by Arbor developers to test communication performance.");
+
+    benchmark_cell
+        .def(pybind11::init<>(
+            [](const regular_schedule_shim& sched, double ratio){
+                return arb::benchmark_cell{sched.schedule(), ratio};}),
+            "schedule"_a, "realtime_ratio"_a)
+        .def(pybind11::init<>(
+            [](const explicit_schedule_shim& sched, double ratio){
+                return arb::benchmark_cell{sched.schedule(), ratio};}),
+            "schedule"_a, "realtime_ratio"_a)
+        .def(pybind11::init<>(
+            [](const poisson_schedule_shim& sched, double ratio){
+                return arb::benchmark_cell{sched.schedule(), ratio};}),
+            "schedule"_a, "realtime_ratio"_a)
+        .def("__repr__", [](const arb::spike_source_cell&){return "<benchmark_cell>";})
+        .def("__str__",  [](const arb::spike_source_cell&){return "<benchmark_cell>";});
```

Comment:
- Add descriptions in constructors. 
- Correct `arb::benchmark_cell` in `__str__` and `__repr__` and add `arbor: ` in return string
- Do we want/need a default for the realtime ratio?

Reply:
1 seems like a reasonable default for the `realtime_ratio`.

Reply:
👍